### PR TITLE
Async: Test Coverage

### PR DESCRIPTION
**Pull request by moizarafat**
## Description
Add more test coverage for Async Module.

## Motivation and Context
Add more test coverage for Async Module.

## How Has This Been Tested?
Existing and New Test Cases.

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: elide-async/src/main/java/com/yahoo/elide/async/export/formatter/CSVExportFormatter.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
     private static final String COMMA = ",";
     private static final String DOUBLE_QUOTES = "\"";
 
-    public static boolean skipCSVHeader = false;
+    public boolean skipCSVHeader = false;
```
Comment (aklish) [inline elide-async/src/main/java/com/yahoo/elide/async/export/formatter/CSVExportFormatter.java:30]: No public member variables.

File: elide-async/src/main/java/com/yahoo/elide/async/export/formatter/CSVExportFormatter.java [hunk 2]
```diff
@@ -82,6 +82,10 @@ public String format(PersistentResource resource, Integer recordNumber) {
      * @return returns Header string which is in CSV format.
      */
     private String generateCSVHeader(EntityProjection projection) {
+        if (projection.getAttributes() == null) {
+            return null;
```
Comment (aklish) [inline elide-async/src/main/java/com/yahoo/elide/async/export/formatter/CSVExportFormatter.java:86]: Empty string?

File: elide-async/src/main/java/com/yahoo/elide/async/operation/GraphQLAsyncQueryOperation.java [hunk 3]
```diff
@@ -35,11 +36,16 @@ public GraphQLAsyncQueryOperation(AsyncExecutorService service, AsyncAPI queryOb
...
         QueryRunner runner = getService().getRunners().get(apiVersion);
+        if (runner == null) {
+            throw new InvalidOperationException("Invalid API Version");
+        }
         UUID requestUUID = UUID.fromString(queryObj.getRequestId());
         //TODO - we need to add the baseUrlEndpoint to the queryObject.
         ElideResponse response = runner.run("", queryObj.getQuery(), user, requestUUID);
-        log.debug("GRAPHQL_V1_0 getResponseCode: {}, GRAPHQL_V1_0 getBody: {}",
-                response.getResponseCode(), response.getBody());
+        if (response != null) {
```
Comment (aklish) [inline elide-async/src/main/java/com/yahoo/elide/async/operation/GraphQLAsyncQueryOperation.java:45]: When is the response null?

File: elide-async/src/main/java/com/yahoo/elide/async/operation/JSONAPIAsyncQueryOperation.java [hunk 4]
```diff
@@ -47,8 +48,10 @@ public ElideResponse execute(AsyncAPI queryObj, User user, String apiVersion)
 
         //TODO - we need to add the baseUrlEndpoint to the queryObject.
         ElideResponse response = elide.get("", getPath(uri), queryParams, user, apiVersion, requestUUID);
-        log.debug("JSONAPI_V1_0 getResponseCode: {}, JSONAPI_V1_0 getBody: {}",
-                response.getResponseCode(), response.getBody());
+        if (response != null) {
```
Comment (aklish) [inline elide-async/src/main/java/com/yahoo/elide/async/operation/JSONAPIAsyncQueryOperation.java:51]: When is the response null?

File: elide-async/src/main/java/com/yahoo/elide/async/operation/JSONAPIAsyncQueryOperation.java [hunk 5]
```diff
@@ -81,7 +84,11 @@ private String getPath(URIBuilder uri) {
     public Integer calculateRecordCount(AsyncQuery queryObj, ElideResponse response) {
         Integer count = null;
         if (response.getResponseCode() == 200) {
-            count = JsonPath.read(response.getBody(), "$.data.length()");
+            try {
+                count = JsonPath.read(response.getBody(), "$.data.length()");
+            } catch (PathNotFoundException e) {
```
Comment (aklish) [inline elide-async/src/main/java/com/yahoo/elide/async/operation/JSONAPIAsyncQueryOperation.java:89]: When is the path not found?

**Review by aklish — CHANGES_REQUESTED** [review]

File: elide-async/src/main/java/com/yahoo/elide/async/operation/JSONAPIAsyncQueryOperation.java (hunk 5 above)
Comment (moizarafat) [reply elide-async/src/main/java/com/yahoo/elide/async/operation/JSONAPIAsyncQueryOperation.java]: removed the logic. It only happens during error scenarios, we will not call it in that situation,

**Review by aklish — APPROVED** [review]

**Pull request merged**